examples/metrics.py

```diff
@@ -29,10 +35,11 @@ def base_type():
 
 class MetricCollector:
 
-    def __init__(self):
+    def __init__(self, datadog_api: DatadogApi):
```

Comment:
We don't really need this to be a constructor argument, since we're always passing a new default instance `DatadogApi()`. Can just do `self.datadog_api = DatadogApi()` on line 42

Reply:
Good catch! Fixed.